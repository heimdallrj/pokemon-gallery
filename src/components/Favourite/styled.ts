import styled from 'styled-components';

import LikeIconSource from 'assets/icons/Favourite';
import likedIconSource from 'assets/icons/Favourite/Filled';

export const LikeIcon = styled(LikeIconSource)`
  width: 16px;
  color: red;
  padding: 0 5px;
`;

export const LikedIcon = styled(likedIconSource)`
  width: 16px;
  color: red;
  padding: 0 5px;
`;
