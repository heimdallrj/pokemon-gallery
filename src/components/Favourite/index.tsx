import { LikedIcon, LikeIcon } from './styled';

type Props = {
  liked?: boolean;
  onClick: () => void;
};

export default function Favourite({ liked, onClick }: Props) {
  return liked ? (
    <LikedIcon onClick={onClick} />
  ) : (
    <LikeIcon onClick={onClick} />
  );
}
