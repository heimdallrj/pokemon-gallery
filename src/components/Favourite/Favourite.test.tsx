import renderer, { act } from 'react-test-renderer';

import Favourite from './index';

describe('Components - Favourite', () => {
  test('renders when both liked or not liked', () => {
    let root: any;
    act(() => {
      root = renderer.create(<Favourite onClick={() => {}} liked={false} />);
    });

    expect(root.toJSON()).toMatchSnapshot();

    // Update likced flag
    act(() => {
      root.update(<Favourite onClick={() => {}} liked />);
    });

    expect(root.toJSON()).toMatchSnapshot();
  });
});
