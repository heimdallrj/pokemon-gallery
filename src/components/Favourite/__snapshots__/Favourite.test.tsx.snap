// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Favourite renders when both liked or not liked 1`] = `
<svg
  className="sc-bdvvtL jbVkAX"
  fill="none"
  onClick={[Function]}
  stroke="currentColor"
  viewBox="0 0 24 24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
  />
</svg>
`;

exports[`Components - Favourite renders when both liked or not liked 2`] = `
<svg
  className="sc-gsDKAQ enRjmm"
  fill="currentColor"
  onClick={[Function]}
  viewBox="0 0 20 20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clipRule="evenodd"
    d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
    fillRule="evenodd"
  />
</svg>
`;
